### PR TITLE
fix(STEF-2475) fix and refactor backtesting iction in context of backtestforecasting config for clarity. Added more colors. Fixed data split function to handle 0.0 splits.

### PR DESCRIPTION
**Pull request by egordm**
This pull request introduces a breaking change by renaming the forecasting configuration fields `horizon_length` and `horizon_min_length` to `predict_length` and `predict_min_length` throughout the codebase. This update affects configuration classes, forecasting logic, backtesting workflows, and associated tests, ensuring consistency and clarity in terminology. Additional improvements include stricter validation of time window selection and minor enhancements to plotting capabilities.

**Configuration and API changes:**

* Renamed `horizon_length` and `horizon_min_length` fields to `predict_length` and `predict_min_length` in `BacktestForecasterConfig` and updated all usages across the codebase, including initialization, event generation, and tests. [[1]](diffhunk://#diff-912e651c76c7aa249fcc52299eb2818993978b4d12e222cc627994df69e5cdf3L38-R39) [[2]](diffhunk://#diff-99b6f9facef55120137081182b76f28a5f068dc8c1be2ce26526a88d794dad9aL105-R106) [[3]](diffhunk://#diff-1531d732dd6abb4a42992a1b1e369004ef29843d3274b2e3518f756f805a1bf0L46-R47) [[4]](diffhunk://#diff-ce2810b090281725c3377c0ae9c32ddd0e8cd2021c72e2ad25dae87db26e83aaL102-R102) [[5]](diffhunk://#diff-ce2810b090281725c3377c0ae9c32ddd0e8cd2021c72e2ad25dae87db26e83aaL127-R127) [[6]](diffhunk://#diff-149e8dd2ccfd0e470162fe39f8ef9dbc2cfbcc12a7c575f8ebb48d513244c0bdL20-R21) [[7]](diffhunk://#diff-149e8dd2ccfd0e470162fe39f8ef9dbc2cfbcc12a7c575f8ebb48d513244c0bdL125-R126) [[8]](diffhunk://#diff-149e8dd2ccfd0e470162fe39f8ef9dbc2cfbcc12a7c575f8ebb48d513244c0bdL168-R169) [[9]](diffhunk://#diff-149e8dd2ccfd0e470162fe39f8ef9dbc2cfbcc12a7c575f8ebb48d513244c0bdL198-R199) [[10]](diffhunk://#diff-c53dc5b83f8b7371f7823b46d221564f539f87eaa49eb801d07df193d65635b2L120-R121) [[11]](diffhunk://#diff-c53dc5b83f8b7371f7823b46d221564f539f87eaa49eb801d07df193d65635b2L176-R177) [[12]](diffhunk://#diff-c53dc5b83f8b7371f7823b46d221564f539f87eaa49eb801d07df193d65635b2L214-R215) [[13]](diffhunk://#diff-c53dc5b83f8b7371f7823b46d221564f539f87eaa49eb801d07df193d65635b2L275-R276) [[14]](diffhunk://#diff-c53dc5b83f8b7371f7823b46d221564f539f87eaa49eb801d07df193d65635b2L338-R339) [[15]](diffhunk://#diff-e0c14625919a7631a61ce3b1a885e60a0cfae74845146350abd8aaca21bb4cf7L28-R29) [[16]](diffhunk://#diff-7b8bc54929f1b1459a47700f3a6d06eb2b7edf4b37cd8625d1d4ffdeb6d6d146L166-R167)

**Backtesting and forecasting logic:**

* Updated logic in `openstef4_backtest_forecaster.py` to use the new field names for training and prediction windows, removed unnecessary conversion steps, and improved clarity in data extraction methods. [[1]](diffhunk://#diff-5b84c7783ee10e4caf4672c869bb908b261201801ed5cb0901074250b05282c4L64-R65) [[2]](diffhunk://#diff-5b84c7783ee10e4caf4672c869bb908b261201801ed5cb0901074250b05282c4L98-L117)
* Improved validation in `RestrictedHorizonVersionedTimeSeries.get_window_versioned` to raise an error if `available_before` is after the horizon, ensuring correct data selection and preventing lookahead bias. [[1]](diffhunk://#diff-bb3c82239a31a3e75a87b6852ba41fce5e3afe3bde23f9751dc19ec6533d4c22R55-R67) [[2]](diffhunk://#diff-bb3c82239a31a3e75a87b6852ba41fce5e3afe3bde23f9751dc19ec6533d4c22L44-R44)

**Testing and reliability:**

* Updated all relevant unit tests to use the new field names and validated correct behavior with the updated configuration. [[1]](diffhunk://#diff-149e8dd2ccfd0e470162fe39f8ef9dbc2cfbcc12a7c575f8ebb48d513244c0bdL20-R21) [[2]](diffhunk://#diff-149e8dd2ccfd0e470162fe39f8ef9dbc2cfbcc12a7c575f8ebb48d513244c0bdL125-R126) [[3]](diffhunk://#diff-149e8dd2ccfd0e470162fe39f8ef9dbc2cfbcc12a7c575f8ebb48d513244c0bdL168-R169) [[4]](diffhunk://#diff-149e8dd2ccfd0e470162fe39f8ef9dbc2cfbcc12a7c575f8ebb48d513244c0bdL198-R199) [[5]](diffhunk://#diff-c53dc5b83f8b7371f7823b46d221564f539f87eaa49eb801d07df193d65635b2L120-R121) [[6]](diffhunk://#diff-c53dc5b83f8b7371f7823b46d221564f539f87eaa49eb801d07df193d65635b2L176-R177) [[7]](diffhunk://#diff-c53dc5b83f8b7371f7823b46d221564f539f87eaa49eb801d07df193d65635b2L214-R215) [[8]](diffhunk://#diff-c53dc5b83f8b7371f7823b46d221564f539f87eaa49eb801d07df193d65635b2L275-R276) [[9]](diffhunk://#diff-c53dc5b83f8b7371f7823b46d221564f539f87eaa49eb801d07df193d65635b2L338-R339) [[10]](diffhunk://#diff-e0c14625919a7631a61ce3b1a885e60a0cfae74845146350abd8aaca21bb4cf7L28-R29) [[11]](diffhunk://#diff-7b8bc54929f1b1459a47700f3a6d06eb2b7edf4b37cd8625d1d4ffdeb6d6d146L166-R167)

**Plotting and visualization:**

* Added new color schemes (`magenta`, `grey`) to the `ForecastTimeSeriesPlotter` for improved visualization options.

**Miscellaneous code improvements:**

* Fixed train/test split logic in `stratified_train_test_split` to correctly assign test days and ensure minimum test set size only when required. [[1]](diffhunk://#diff-de0a58caa2041948326aed0027aa78f8f008344dbf81676557472ea2dd67b5a0L158-R160) [[2]](diffhunk://#diff-de0a58caa2041948326aed0027aa78f8f008344dbf81676557472ea2dd67b5a0L169-R177)
* Removed unused fields from `TargetProvider` and cleaned up related code.
* Removed unused import from `openstef4_backtest_forecaster.py`.

### CODE REVIEW

File: packages/openstef-beam/src/openstef_beam/backtesting/backtest_forecaster/openstef4_backtest_forecaster.py [hunk 1]
```diff
@@ -61,20 +60,9 @@ def fit(self, data: RestrictedHorizonVersionedTimeSeries) -> None:
         # Create a new workflow for this training cycle
         self._workflow = self.workflow_factory()
 
-        # Get training data window based on config
-        training_end = data.horizon
-        training_start = training_end - self.config.training_context_length
-
         # Extract the versioned dataset for training
```
Comment (MvLieshout) [inline packages/openstef-beam/src/openstef_beam/backtesting/backtest_forecaster/openstef4_backtest_forecaster.py]:
```suggestion
        # Extract the dataset for training
```
Not versioned anymore right?

File: packages/openstef-beam/src/openstef_beam/backtesting/backtest_forecaster/openstef4_backtest_forecaster.py [hunk 2]
```diff
@@ -95,26 +83,13 @@ def predict(self, data: RestrictedHorizonVersionedTimeSeries) -> TimeSeriesDatas
...
         # Extract the dataset including both historical context and forecast period
-        predict_data_versioned = data.get_window_versioned(
-            start=predict_context_start,
-            end=forecast_end,  # Include the forecast period
+        predict_data = data.get_window(
+            start=data.horizon - self.config.predict_context_length,
+            end=data.horizon + self.config.predict_length,  # Include the forecast period
             available_before=data.horizon,  # Only use data available at prediction time (prevents lookahead bias)
         )
         # Convert to horizons
```
Comment (MvLieshout) [inline packages/openstef-beam/src/openstef_beam/backtesting/backtest_forecaster/openstef4_backtest_forecaster.py]:
```suggestion
```

**Review by MvLieshout — APPROVED** [review]
Also some small nitpicks, but only in the comments.

File: packages/openstef-beam/src/openstef_beam/backtesting/backtest_forecaster/openstef4_backtest_forecaster.py (hunk 1 above)
Comment (egordm) [reply packages/openstef-beam/src/openstef_beam/backtesting/backtest_forecaster/openstef4_backtest_forecaster.py]: You are right. Fixed.

File: packages/openstef-beam/src/openstef_beam/backtesting/backtest_forecaster/openstef4_backtest_forecaster.py (hunk 2 above)
Comment (egordm) [reply packages/openstef-beam/src/openstef_beam/backtesting/backtest_forecaster/openstef4_backtest_forecaster.py]: Fixed. Thx
